lgsm/functions/workshop_core.sh

```diff
@@ -0,0 +1,318 @@
+		echo -en "Found ${filesupperwc} uppercase files out of ${fileswc}, converting..."
+		#
+		# Coudln't get this to work on WSL. Needs to be verified on an acutal linux server.
+		#
+		# Convert files and directories starting from the deepest to prevent issues (-depth argument)
+		while IFS= read -r -d '' src; do
+			# We have to convert only the last file from the path, otherwise we will fail to convert anything if a parent dir has any uppercase
+			# therefore, we have to separate the end of the filename to only lowercase it rather than the whole line
+			# Gather parent dir, filename lowercase filename, and set lowercase destination name
+			latestparentdir=$(dirname "${src}")
+			latestfilelc=$(basename "${src}" | tr '[:upper:]' '[:lower:]')
+			dst="${latestparentdir}/${latestfilelc}"
+			# Only convert if destination does not already exist for some reason
+			if [ ! -e "${dst}" ]; then
+				# Finally we can rename the file
+				mv "${src}" "${dst}"
+				# Exit if it fails for any reason
+				local exitcode=$?
+				if [ "${exitcode}" != 0 ]; then
+					fn_print_fail_eol_nl
+					core_exit.sh
+				fi
+			fi
+		done < <(find "${workshopmodsdir}" -depth -name '*[[:upper:]]*' -print0)
+		fn_print_ok_eol_nl
+	fi
+}
+
+# # Copy the mod into serverfiles.
+fn_workshop_copy_destination() {
+	local modid="$1"
+	local modname="$(fn_workshop_get_mod_name ${modid})"
+	if fn_workshop_is_mod_copy_needed ${modid}; then
+		echo "Copying mod ${modname} (${modid})"
+		# If workshop mod exists in installation folder, delete it for clean install
+		if [ -d "${workshopmodsdir}/${modid}" ]; then
+			rm -rf "${workshopmodsdir}/${modid}"
+		fi
+		modsrc="${workshopmodsdldir}/steamapps/workshop/content/${gameappid}/${modid}"
+		cp -fa ${modsrc} ${workshopmodsdir}
+		if [ "${engine}" == "realvirtuality" ]; then
+			modkey="${workshopmodsdldir}/steamapps/workshop/content/${gameappid}/${modid}/keys"
+			if ! [ -d "${modkey}" ]; then
+				modkey="$steamcmd/steamapps/workshop/content/${gameappid}/${modid}/Keys"
+			fi
+			if ! [ -d "${modkey}" ]; then
+				modkey="$steamcmd/steamapps/workshop/content/${gameappid}/${modid}/key"
+			fi
+			if ! [ -d "${modkey}" ]; then
+				modkey="$steamcmd/steamapps/workshop/content/${gameappid}/${modid}/Key"
+			fi
+			if ! [ -d "${modkey}" ]; then
+				echo "Mod ${modname} seems to be missing key folder. Tring to copy key from the main folder."
+				cp -fa "${workshopmodsdir}/${modid}/*.bikey" ${keysdir}
+			else
+				cp -fa ${modkey}/*.bikey ${keysdir}
+			fi
+		fi
+	else
+		echo "Mod ${modname}  is already in mods folder."
+	fi
+}
+
+# ## Directory management.
+
+# Create mods files and directories if it doesn't exist.
+fn_create_workshop_dir() {
+	# Create lgsm data modsdir.
+	if [ ! -d "${workshopmodsdldir}" ]; then
+		echo -en "creating LinuxGSM Steam Workshop data directory ${workshopmodsdldir}..."
+		mkdir -p "${workshopmodsdldir}"
+		exitcode=$?
```

Comment:
This is redundant. Sometimes it's useful to specifically catch `$?`, but this is mainly when you're testing for something other than just 0 or non-0, in my experience.

In this case, it's easier and more practical to directly act upon the exit status, rather than separately store the status, check the value, then act based on the value, especially if you're using `[` (inefficient). The `if` statement in shell programming essentially does this for us.

For example:

```bash
if mkdir -p "${workshopmodsdldir}"; then
    ...
else
    ...
fi
```

While this isolated case isn't a huge deal, a lot of redundant code can be a problem, both to readability and performance.

This is core BASH functionality.